### PR TITLE
ci: 1.79 fixes (replace `macos-latest stable` with `macos-latest 1.78`)

### PR DESCRIPTION
**Pull request by dj8yfo**
this passes if `macos-latest` is temporarily commented out https://github.com/near/near-sdk-rs/actions/runs/9566693799 ;
otherwise, `macos-latest` is having some serious trouble on 1.79

### CODE REVIEW

**Comment by ruseinov** [comment]
I have this already fixed in my working branch.

**Comment by ruseinov** [comment]
I'm looking into this in another branch and locally, if I make progress - I will let you know.

**Comment by ruseinov** [comment]
This is not reproducible locally on my mac. Could it be the cache?

**Comment by dj8yfo** [comment]
@ruseinov , what's your processor type? It might be some wicked bug of 1.79 affecting only `aarch64-*-*`, or it might be not

**Comment by ruseinov** [comment]
> @ruseinov , what's the processor type of your machine? It might be some wicked bug of 1.79 affecting only `aarch64-*-*`, or it might be not

I have an m2-pro, so that should have popped up.

**Comment by ruseinov** [comment]
I have tried investigating this with `macos-13` - that yields "test panicked while processing panic". There might indeed be something wrong with the setup somewhere.

**Comment by ruseinov** [comment]
I guess you have already talked to @frol. Perhaps we should pin the version to 1.78 as a temp fix, either via `rust-toolchain.toml` or just via CI.

**Comment by dj8yfo** [comment]
@ruseinov created an [issue](https://github.com/rust-lang/rust/issues/126643), as my somewhat limited research hasn't been particularly useful

**Comment by ruseinov** [comment]
> @ruseinov created an [issue](https://github.com/rust-lang/rust/issues/126643), as my somewhat limited research hasn't been particularly useful

That’s fine, let’s so 1.78 to fix master for now. And solve the issue later, unless it solves itself with the next version. Can’t be debugging machine-specific failures with actual access to the machine.

File: near-sdk-macros/src/core_impl/info_extractor/arg_info.rs [hunk 1]
```diff
@@ -18,12 +18,15 @@ pub enum BindgenArgType {
 /// A single argument of a function after it was processed by the bindgen.
 pub struct ArgInfo {
     /// Attributes not related to bindgen.
+    #[allow(unused)]
```
Comment (frol) [inline near-sdk-macros/src/core_impl/info_extractor/arg_info.rs:21]: Are these really unused or is it a clippy bug?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -8,26 +8,27 @@ env:
...
   test:
-    runs-on: ${{ matrix.platform }}
-    name: "${{ matrix.platform }} ${{ matrix.toolchain }}"
+    runs-on: ${{ matrix.platform.os }}
+    name: "${{ matrix.platform.os }} ${{ matrix.platform.rs }}"
     strategy:
       matrix:
-        platform: [ubuntu-latest, macos-latest]
-        toolchain: [stable, 1.76.0]
+        platform: [{os: ubuntu-latest, rs: 1.76.0}, {os: ubuntu-latest, rs: stable}, {os: macos-latest, rs: 1.76.0}, {os: macos-latest, rs: 1.78.0}]
```
Comment (frol) [inline .github/workflows/test.yml:15]: I think this should work and it is easier to read and maintain:

```suggestion
        platform:
          - os: ubuntu-latest
            rs: 1.76.0
          - os: ubuntu-latest
            rs: stable
          - os: macos-latest
            rs: 1.76.0
          - os: macos-latest
            rs: 1.78.0
```

**Review by frol — DISMISSED** [review]

File: near-sdk-macros/src/core_impl/info_extractor/arg_info.rs (hunk 1 above)
Comment (dj8yfo) [reply near-sdk-macros/src/core_impl/info_extractor/arg_info.rs:21]: as far as go-to-references have told me, these are unused. looks like the code was copied from elsewhere, 
 but only some of the fields are used in the context of  `near-sdk-macros`

Comment (dj8yfo) [reply near-sdk-macros/src/core_impl/info_extractor/arg_info.rs:21]: triple-checked these; they are only created inside of `::new` constructors, and not touched by other code thereafter. So, as far as tooling can tell, these are unused

**Review by frol — APPROVED** [review]

**Pull request merged**